core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -5192,6 +5192,25 @@ private void checkNullOperand(SqlTester tester, String op) {
         "07000000789c4bad48cc2dc84905000bc002ed", "VARBINARY NOT NULL");
   }
 
+  @Test public void testUncompress() {
+    SqlTester sqlTester = tester(SqlLibrary.MYSQL);
+    sqlTester.checkNull("UNCOMPRESS(NULL)");
+    sqlTester.checkString("UNCOMPRESS(x'')", "", "VARCHAR");
+
+    sqlTester.checkNull("UNCOMPRESS(x'1233')");
+
+    sqlTester.checkString("UNCOMPRESS(COMPRESS('test'))",
+        "test", "VARCHAR");
+
+    sqlTester.checkString("UNCOMPRESS(x'10000000789c4b4c44050033980611')",
+        "aaaaaaaaaaaaaaaa", "VARCHAR");
+
+    sqlTester.checkString("UNCOMPRESS(x'06000000789c2b4ecc2dc849050008de0283' )",
+        "sample", "VARCHAR");
+    sqlTester.checkString("UNCOMPRESS(x'07000000789c4bad48cc2dc84905000bc002ed')",
+        "example", "VARCHAR");
+  }
```

Comment:
Can add tests based on error？
`sqlTester.checkFails("UNCOMPRESS('test')","(?s).*Only ByteString.*", true);`

Reply:
Done :)

Reply:
> Done :)

My fault, I listed an example. You need to increase the corresponding error message capture.

Reply:
@XuQianJin-Stars After master merge I'm facing below error 
`[Error while executing SQL "select UNCOMPRESS(p0) from (values ('test')) as t(p0)": Error while compiling generated Java code:`
Which makes it difficult to write test case
` sqlTester.checkFails("UNCOMPRESS('test')","(?s).*Only ByteString.*", true);`
So I have removed it for now, Also as per contract only bytestring is allowed as an argument, Not sure if we really need this test case

Reply:
What is the error Java code like ?

Reply:
`rror while executing SQL "select UNCOMPRESS(p0) from (values ('test')) as t(p0)": Error while compiling generated Java code:
public org.apache.calcite.linq4j.Enumerable bind(final org.apache.calcite.DataContext root) {
  final org.apache.calcite.linq4j.Enumerable _inputEnumerable = org.apache.calcite.linq4j.Linq4j.asEnumerable(new String[] {
    "test"});
  return new org.apache.calcite.linq4j.AbstractEnumerable(){
      public org.apache.calcite.linq4j.Enumerator enumerator() {
        return new org.apache.calcite.linq4j.Enumerator(){
            public final org.apache.calcite.linq4j.Enumerator inputEnumerator = _inputEnumerable.enumerator();
            public void reset() {
              inputEnumerator.reset();
            }

            public boolean moveNext() {
              return inputEnumerator.moveNext();
            }

            public void close() {
              inputEnumerator.close();
            }

            public Object current() {
              return org.apache.calcite.runtime.CompressionFunctions.uncompress((org.apache.calcite.avatica.util.ByteString) (inputEnumerator.current() == null ? (String) null : inputEnumerator.current().toString()));
            }

          };
      }

    };
}


public Class getElementType() {
  return java.lang.String.class;
}


];`

Reply:
The query should fail in validation phase.
It is too late to throw the exception in codegen phase.

Reply:
So should we unblock this feature or resolve validation issue first?

Reply:
I'm inclined to fix the validation issue first. If the problem can be reproduced outside of this PR, we can file another JIRA.

Reply:
MySQL Supports string values in uncompress function which returns NULL. Please refer: 

`mysql> SELECT UNCOMPRESS('string');
+----------------------+
| UNCOMPRESS('string') |
+----------------------+
| NULL                 |
+----------------------+
1 row in set, 1 warning (0.00 sec)
`

So, This should not be failure test case

Reply:
I guess MySQL may trigger the implicit type coercion for `STRING` params.

Reply:
@danny0405 Can you please look for attached error, and help me understand the issue.

Reply:
The implicit type coercion did triggered for `UNCOMPRESS('string')` during SQL validation, but the runtime code gen does not do it well. I suggest to move the `UNCOMPRESS('string')` test to `SqlValidatorTest`, a data type validation test is enough.

We can add the operator test if we really support the `CAST(VARCHAR as BINARY)` in runtime. Maybe @DonnyZone  can give a help.

Reply:
The problem in runtime lies in: `org.apache.calcite.avatica.util.ByteString) xxx`
This piece of code is generated by `EnumUtils.convert`. If we want to fix it, we need to add the converting logic (for `String` to `ByteString`) in `EnumUtils.convert`.
I am just wondering how to implement a general logic? `null` or `Exception` or other implementations?

Reply:
For Apache Spark and Flink, it is just `string.getBytes()`, but for MySQL, PostgreSQL and MS-SQL, the outputs is different, i didn't have time to do a deeper research, maybe you can have a try.

Reply:
@danny0405 @DonnyZone Did quick search and found MySQL outputs text as is and MSQSL gives output of what String.getBytes gives. How should we proceed?